### ISSUE TITLE
404 instead of setup screen

### ISSUE THREAD

**Issue by schaten** [jacobalberty/unifi-docker#510]
### Operating system

Debian 11.2

### UniFi Tag

latest

### Docker run

podman run --rm --ip=172.16.16.2 -e TZ=America/Pacific -e RUNAS_UID0=false -e UNIFI_UID=1001 -e UNIFI_GID=1001 -v $PWD/unifi-vol:/unifi:Z -v $PWD/run:/var/run/unifi:Z --name unifi docker.io/jacobalberty/unifi:latest

### Bug description

Opening https://localhost:8443/ in a browser gives
HTTP Status 404 – Not Found.

Sometimes it also shows the initial setup screen. Very unpredictable.
In contrast, https://localhost:8443/setup seems to work always, so far.

Actually I think this might just as well be a problem with Unifi Controller software, not the image. In this case apologies for reporting it here - still maybe someone else has this problem.
It also seems similar to #385 .

### Steps to reproduce

On a fresh Debian install, as root user:
```
apt install podman
useradd unifi
mkdir unifi unifi/unifi-vol unifi/run
cd unifi
podman run --rm --ip=172.16.16.2 -e TZ=America/Pacific -e RUNAS_UID0=false -e UNIFI_UID=1001 -e UNIFI_GID=1001 -v $PWD/unifi-vol:/unifi:Z -v $PWD/run:/var/run/unifi:Z --name unifi docker.io/jacobalberty/unifi:latest
```

### Relevant log output

```shell
*nothin* suspicious as far as I can tell.

server.log:
[2022-01-02T22:07:53,992] <launcher> WARN  system - reload system.properties failed: file not found
[2022-01-02T22:07:53,998] <launcher> WARN  system - reload system.properties.bk failed: file not found
[2022-01-02T22:07:53,999] <launcher> INFO  system - *** Running for the first time, creating identity ***
[2022-01-02T22:07:53,999] <launcher> INFO  system - UUID: 21901b5b-7666-435b-8901-2eed934343b1
[2022-01-02T22:07:53,999] <launcher> WARN  system - reload system.properties failed: file not found
[2022-01-02T22:07:53,999] <launcher> WARN  system - reload system.properties.bk failed: file not found
[2022-01-02T22:07:54,005] <launcher> INFO  system - Reporter UUID: aa123c7c-aa42-445d-bedb-d41f62efe0f8
[2022-01-02T22:07:54,504] <launcher> WARN  system - Valid keystore is missing or invalid. Generating one ...
[2022-01-02T22:07:54,504] <launcher> INFO  system - Generating Certificate[UniFi]... please wait...
[2022-01-02T22:07:55,005] <launcher> INFO  system - Certificate[UniFi] generated!
[2022-01-02T22:07:55,048] <launcher> INFO  tomcat - Adding basic REST API support during the startup
[2022-01-02T22:07:55,681] <launcher> INFO  system - Tomcat startup took 1566ms
[2022-01-02T22:07:55,682] <launcher> INFO  db     - New installation, skipping migration
[2022-01-02T22:07:55,699] <launcher> INFO  tomcat - Adding basic REST API support during the startup
[2022-01-02T22:07:56,779] <launcher> INFO  system - ======================================================================
[2022-01-02T22:07:56,779] <launcher> INFO  system - UniFi 6.5.55 (build atag_6.5.55_16678 - release/release) is started
[2022-01-02T22:07:56,779] <launcher> INFO  system - ======================================================================
[2022-01-02T22:07:56,780] <launcher> INFO  system - BASE dir:/usr/lib/unifi
[2022-01-02T22:07:56,827] <launcher> INFO  system - Current System IP: 172.16.16.2
[2022-01-02T22:07:56,827] <launcher> INFO  system - Hostname: 1c930c4abf7c
[2022-01-02T22:07:56,827] <launcher> INFO  system - ubic.env: prod
[2022-01-02T22:07:56,978] <launcher> INFO  db     - waiting for db connection...
[2022-01-02T22:07:58,143] <db-server> INFO  db     - 2022-01-02T22:07:58.142+0000 I STORAGE  [main] Engine custom option: cache_size=256M
[2022-01-02T22:07:58,479] <launcher> INFO  db     - Connecting to mongodb://localhost:27117

mongod.log:
2022-01-02T22:07:58.732+0000 I CONTROL  [initandlisten] MongoDB starting : pid=92 port=27117 dbpath=/usr/lib/unifi/data/db 64-bit host=1c930c4abf7c
2022-01-02T22:07:58.732+0000 I CONTROL  [initandlisten] db version v3.6.3
2022-01-02T22:07:58.732+0000 I CONTROL  [initandlisten] git version: 9586e557d54ef70f9ca4b43c26892cd55257e1a5
2022-01-02T22:07:58.732+0000 I CONTROL  [initandlisten] OpenSSL version: OpenSSL 1.1.1  11 Sep 2018
2022-01-02T22:07:58.732+0000 I CONTROL  [initandlisten] allocator: tcmalloc
2022-01-02T22:07:58.732+0000 I CONTROL  [initandlisten] modules: none
2022-01-02T22:07:58.732+0000 I CONTROL  [initandlisten] build environment:
2022-01-02T22:07:58.732+0000 I CONTROL  [initandlisten]     distarch: x86_64
2022-01-02T22:07:58.732+0000 I CONTROL  [initandlisten]     target_arch: x86_64
2022-01-02T22:07:58.732+0000 I CONTROL  [initandlisten] options: { net: { bindIp: "127.0.0.1", port: 27117, unixDomainSocket: { pathPrefix: "/usr/lib/unifi/run" } }, processManagement: { pidFilePath: "/usr/lib/unifi/run/mongod.pid" }, storage: { dbPath: "/usr/lib/unifi/data/db", wiredTiger: { engineConfig: { configString: "cache_size=256M" } } }, systemLog: { destination: "file", logAppend: true, logRotate: "reopen", path: "/usr/lib/unifi/logs/mongod.log" } }
2022-01-02T22:07:58.733+0000 I STORAGE  [initandlisten] 
2022-01-02T22:07:58.733+0000 I STORAGE  [initandlisten] ** WARNING: Using the XFS filesystem is strongly recommended with the WiredTiger storage engine
2022-01-02T22:07:58.733+0000 I STORAGE  [initandlisten] **          See http://dochub.mongodb.org/core/prodnotes-filesystem
2022-01-02T22:07:58.733+0000 I STORAGE  [initandlisten] wiredtiger_open config: create,cache_size=479M,session_max=20000,eviction=(threads_min=4,threads_max=4),config_base=false,statistics=(fast),log=(enabled=true,archive=true,path=journal,compressor=snappy),file_manager=(close_idle_time=100000),statistics_log=(wait=0),verbose=(recovery_progress),cache_size=256M
2022-01-02T22:08:00.023+0000 I CONTROL  [initandlisten] 
2022-01-02T22:08:00.023+0000 I CONTROL  [initandlisten] ** WARNING: Access control is not enabled for the database.
2022-01-02T22:08:00.023+0000 I CONTROL  [initandlisten] **          Read and write access to data and configuration is unrestricted.
2022-01-02T22:08:00.024+0000 I CONTROL  [initandlisten] 
2022-01-02T22:08:00.024+0000 I CONTROL  [initandlisten] 
2022-01-02T22:08:00.024+0000 I CONTROL  [initandlisten] ** WARNING: /sys/kernel/mm/transparent_hugepage/enabled is 'always'.
2022-01-02T22:08:00.024+0000 I CONTROL  [initandlisten] **        We suggest setting it to 'never'
2022-01-02T22:08:00.024+0000 I CONTROL  [initandlisten] 
2022-01-02T22:08:00.024+0000 I STORAGE  [initandlisten] createCollection: admin.system.version with provided UUID: 9f8f36ed-bc27-4d81-bc03-67bc059ff9c0
2022-01-02T22:08:00.184+0000 I COMMAND  [initandlisten] setting featureCompatibilityVersion to 3.6
2022-01-02T22:08:00.189+0000 I STORAGE  [initandlisten] createCollection: local.startup_log with generated UUID: 36174245-52b0-4bc8-b424-b8f33228a3fe
2022-01-02T22:08:00.393+0000 I FTDC     [initandlisten] Initializing full-time diagnostic data capture with directory '/usr/lib/unifi/data/db/diagnostic.data'
2022-01-02T22:08:00.394+0000 I NETWORK  [initandlisten] waiting for connections on port 27117
2022-01-02T22:13:00.394+0000 I STORAGE  [thread1] createCollection: config.system.sessions with generated UUID: 352b736d-4b7c-42f3-b58a-71d7bee4287f
2022-01-02T22:13:00.643+0000 I INDEX    [thread1] build index on: config.system.sessions properties: { v: 2, key: { lastUse: 1 }, name: "lsidTTLIndex", ns: "config.system.sessions", expireAfterSeconds: 1800 }
2022-01-02T22:13:00.643+0000 I INDEX    [thread1] 	 building index using bulk method; build may temporarily use up to 500 megabytes of RAM
2022-01-02T22:13:00.644+0000 I INDEX    [thread1] build index done.  scanned 0 total records. 0 secs
2022-01-02T22:13:00.644+0000 I COMMAND  [thread1] command config.$cmd command: createIndexes { createIndexes: "system.sessions", indexes: [ { key: { lastUse: 1 }, name: "lsidTTLIndex", expireAfterSeconds: 1800 } ], $db: "config" } numYields:0 reslen:98 locks:{ Global: { acquireCount: { r: 1, w: 1 } }, Database: { acquireCount: { W: 1 } }, Collection: { acquireCount: { w: 1 } } } protocol:op_msg 250ms
```

**Comment by jacobalberty** [comment]
I don't see anything exposing a port. Are you doing this behing traefik or something?

**Comment by schaten** [comment]
Actually I'm doing a `ssh -L 8443:172.16.16.2:8443 containerhost`, just for testing at the moment... Sorry for the confusion.

**Comment by schaten** [comment]
I just wanted to add that I tried a few more things: First, I used a macvlan network, so that I could directly access the container. Furthermore I updated to Debian Bookworm, which has podman 3.4.4, which is newer than the podman on the fedora installation where I tried the image and it just worked™. but no luck.
Furthermore I:
- unloaded all the apparmor profiles
- used `--privileged=true`
- ran the unifi controller as root.
Nothing.
Also, what initially worked for me, appending a /support to the URL, did nothing this weekend... - must've been a coincidence.

It seems that the LinuxServer io image also does not work on some platforms: https://github.com/linuxserver/docker-unifi-controller/issues/100 - which I can confirm. Might be the same reason?

Very confusing.
